src/mame/taito/qix.h

```diff
@@ -39,140 +34,149 @@ class qix_state : public driver_device
 		m_audiocpu(*this, "audiocpu"),
 		m_videocpu(*this, "videocpu"),
 		m_crtc(*this, "vid_u18"),
-		m_pia0(*this, "pia0"),
-		m_pia1(*this, "pia1"),
-		m_pia2(*this, "pia2"),
-		m_sndpia0(*this, "sndpia0"),
-		m_sndpia1(*this, "sndpia1"),
-		m_sndpia2(*this, "sndpia2"),
+		m_pia(*this, "pia%u", 0U),
+		m_sndpia(*this, "sndpia%u", 0U),
 		m_discrete(*this, "discrete"),
-		m_paletteram(*this, "paletteram"),
+		m_screen(*this, "screen"),
+		m_palette(*this, "palette"),
 		m_videoram(*this, "videoram", 0x10000, ENDIANNESS_BIG),
 		m_videoram_address(*this, "videoram_addr"),
-		m_videoram_mask(*this, "videoram_mask"),
 		m_scanline_latch(*this, "scanline_latch"),
-		m_bank0(*this, "bank0"),
-		m_bank1(*this, "bank1"),
-		m_screen(*this, "screen")
+		m_videobank(*this, "videobank")
 	{ }
 
 	void qix_base(machine_config &config);
 	void qix(machine_config &config);
 	void qix_video(machine_config &config);
 	void qix_audio(machine_config &config);
-	void kram3(machine_config &config);
-	void kram3_video(machine_config &config);
```

Comment:
The ones that aren’t complete systems (`qix_video` and `qix_audio`) should be in the `protected` section.  Also, you can put `ATTR_COLD` on machine configuration functions.